CRM/Contact/BAO/Contact.php

```diff
@@ -2029,14 +2029,43 @@ public static function createProfileContact(
     }
 
     // Process group and tag
+    // @todo Contact::create also calls addContactsToGroup/removeContactsToGroup
+    //   Remove from here and use the existing functionality in Contact::create
     if (isset($params['group'])) {
       $method = 'Admin';
       // this for sure means we are coming in via profile since i added it to fix
       // removing contacts from user groups -- lobo
       if ($visibility) {
         $method = 'Web';
       }
-      CRM_Contact_BAO_GroupContact::create($params['group'], $contactID, $visibility, $method);
+      $contactIds = [$contactID];
+      $contactGroup = [];
+
+      if ($contactID) {
+        $contactGroupList = CRM_Contact_BAO_GroupContact::getContactGroup($contactID, 'Added',
+          NULL, FALSE, $visibility
+        );
+        if (is_array($contactGroupList)) {
+          foreach ($contactGroupList as $key) {
+            $groupId = $key['group_id'];
+            $contactGroup[$groupId] = $groupId;
+          }
+        }
+      }
+      // get the list of all the groups
+      $allGroup = CRM_Contact_BAO_GroupContact::getGroupList(0, $visibility);
+
+      // check which values has to be add/remove contact from group
+      foreach ($allGroup as $key => $varValue) {
+        if (!empty($params[$key]) && !array_key_exists($key, $contactGroup)) {
```

Comment:
params that was sent to `GroupContact::create()` was `$params['group']`, so it seems the above check should also be for `$params['group'][$key]`? 

or probably better to have eg `$groupParams = $params['group'] ?? []` and use that instead.

Reply:
Have added a test https://github.com/civicrm/civicrm-core/commit/87d74984d11084418256fec2f220e91bda0e3b0c but it fails due to the above problem. Should pass once updated.

Reply:
Thanks @jitendrapurohit I've updated the PR to fix the params